FrontEnd/src/components/ProfilePage/FormComponents/AdditionalInfo.js

```diff
@@ -42,74 +26,73 @@ const AdditionalInfo = (props) => {
         } else {
             setFoundationYearError(`Рік заснування не в діапазоні 1800-${currentYear}`);
         }
-        setUser((prevState) => {
+        setProfile((prevState) => {
             return { ...prevState, [e.target.name]: e.target.value };
         });
     };
 
     const validateForm = () => {
         let isValid = true;
         const currentYear = new Date().getFullYear();
-        const year = Number(user.foundationYear);
-        if ((1800 > year || year > currentYear) && user.foundationYear) {
+        const year = Number(profile.founded);
+        if ((1800 > year || year > currentYear) && profile.founded) {
             isValid = false;
         }
         return isValid;
     };
 
-    const handleSubmit = (event) => {
+    const handleSubmit = async (event) => {
         event.preventDefault();
         if (validateForm()) {
-            props.onUpdate(user);
-            // TODO something
-        } else {
-            // TODO something
+            const token = localStorage.getItem('Token');
+            try {
+                const response = await fetch(`${process.env.REACT_APP_BASE_API_URL}/api/profiles/${user.profile_id}`, {
```

Comment:
I followed this example https://github.com/itsfaqih/react-swr-auth/tree/main/src . And it was very convenient because I had the opportunity to use mutate, which solved problems and blockers of categories and activities. We have many serializers for different permissions and requests, and the categories` and activities` lists had different structures depending on the above conditions. So, when I tried using mutate and it worked for me, I decided to leave it.

Reply:
Mutate is a good choice if you need to change the data. Fetch and Axios are used for the same goal - to submit API requests. For consistency, I recommend using one approach instead of  mixing them